scripts/ci/maintenance/ghcr_prune/api.py

```
@@ -0,0 +1,198 @@
+    """Extract the ``rel="next"`` URL from a GitHub Link header.
+
+    Args:
+        link_header: Raw ``Link`` header value.
+
+    Returns:
+        Next-page URL or ``None``.
+    """
+    if not link_header:
+        return None
+    for part in link_header.split(","):
+        part = part.strip()
```

Ruff (0.15.12)

[warning] `for` loop variable `part` overwritten by assignment target

(PLW2901)